src/tests/baseservices/compilerservices/UnsafeAccessors/UnsafeAccessorsTests.cs

```diff
@@ -215,6 +217,18 @@ public static void Verify_AccessFieldClass()
         extern static ref string GetPrivateField(UserDataClass d);
     }
 
+    [Fact]
+    public static void Verify_AccessFieldOffsetClass()
+    {
+        Console.WriteLine($"Running {nameof(Verify_AccessFieldOffsetClass)}");
+
+        // Offset = method table pointer + first string pointer
+        Assert.Equal(IntPtr.Size + IntPtr.Size, GetPrivateFieldOffset());
+
+        [UnsafeAccessor(UnsafeAccessorKind.FieldOffset, Name=UserDataClass.FieldSecondName)]
+        extern static nint GetPrivateFieldOffset(UserDataClass? d = null);
```

Comment:
Not sure what others' opinions on this are, but I like to be explicit with these sorts of APIs - I'd like it if we could also specify a second parameter which is the type of the field - it would first check for fields with same type (including modifiers), and then same type (ignoring modifiers) (same as other checks for the `UnsafeAccessor`s iirc).